src/app/common/paging/pageable-table/pageable-table.component.ts

```diff
@@ -15,38 +15,41 @@ import { PagingOptions, PageChange } from '../pager/pager.component';
 })
 export class PageableTable implements AfterContentInit {
 
-	@Input() items: Array<any>;
 	@Input() pagingOptions: PagingOptions = new PagingOptions();
+	@Input() items: Array<any>;
+	@Input() hasItems: boolean = false;
```

Comment:
I would probably make `hasItems` a method that checks if `this.items` is empty or not instead of letting the parent container decide if items exist out-of-band with the items array.

Doing this also will let us call `hasItems()` instead of using `{{ items?.length === 0 }}` in the template. `items?.length` will actually resolve to `null` in the case that items comes back as null/undefined, which will give us the unintended consequence (`null !== 0`) of unhiding all the stuff that's hiding behind that check.

Reply:
`hasItems` isn't intended to mirror the current state of `items`.  It is supposed to represent the initial state.  It is needed so that we can varying the empty state message based on if there were truly no results or if the user applied an additional filter/search that yielded no results.